code/z_adventurezones/space_casino.dm

```diff
@@ -0,0 +1,371 @@
+		playsound(get_turf(src), "[win_sound]", 55, 1)
+		src.working = 0
+		src.icon_state = "slotsitem-off"
+		src.uses++
+
+	emag_act(var/mob/user, var/obj/item/card/emag/E) // Freak out and die
+		src.icon_state = "slotsitem-malf"
+		playsound(get_turf(src), "sound/misc/klaxon.ogg", 55, 1)
+		src.visible_message("<span class='subtle'><b>[src]</b> says, 'WINNER! WINNER! JACKPOT! WINNER! JACKPOT! BIG WINNER! BIG WINNER!'</span>")
+		playsound(src.loc, "sound/impact_sounds/Metal_Clang_1.ogg", 60, 1, pitch = 1.2)
+		animate_shake(src,7,5,2)
+		sleep(3.5 SECONDS)
+
+		src.visible_message("<span class='subtle'><b>[src]</b> says, 'BIG WINNER! BIG WINNER!'</span>")
+		playsound(src.loc, "sound/impact_sounds/Metal_Clang_2.ogg", 60, 1, pitch = 0.8)
+		animate_shake(src,5,7,2)
+		sleep(1.5 SECONDS)
+
+		new/obj/decal/implo(src.loc)
+		playsound(src, 'sound/effects/suck.ogg', 60, 1)
+		if (src.scan)
+			src.scan.set_loc(src.loc)
+		qdel(src)
+
+// Barbuddy [my beloved]
+
+/obj/machinery/bot/barbuddy
+	name = "BarBuddy"
+	desc = "A little bartending robot!"
+	icon = 'icons/obj/bots/aibots.dmi'
+	icon_state = "robuddy1"
+	layer = 5.0 //TODO LAYER
+	density = 0
+	anchored = 0
+	bot_move_delay = FLOORBOT_MOVE_SPEED
+	var/hasDrink = 0
+	var/atom/home // Initialized early. Where the barbuddy should be serving. Barbuddy explodes if taken too far from here.
+	var/list/homeTables = list() // Initialized early. All nearby tables that the barbuddy should be checking for drinks.
+	var/list/targets = list() // Nearby tables that are in need of drinks.
+	var/target
+	var/atom/moveTowards // The object that should be moved towards.
+
+	var/possible_drinks = list("bilk","beer","cider","mead","wine","champagne","rum","vodka","bourbon", \
+							"boorbon","beepskybeer","screwdriver","bloody_mary","bloody_scary",\
+							"snakebite","diesel","suicider","port","gin","vermouth","bitters","whiskey_sour",\
+							"daiquiri","martini","v_martini","murdini","manhattan","libre","ginfizz","gimlet",\
+							"v_gimlet","w_russian","b_russian","irishcoffee","cosmo","beach","gtonic","vtonic","sonic",\
+							"gpink","eraser","squeeze","hunchback","madmen","planter","maitai","harlow",\
+							"gchronic","margarita","tequini","pfire","bull","longisland","pinacolada","longbeach",\
+							"mimosa","french75","sangria","tomcollins","peachschnapps","moscowmule","tequilasunrise",\
+							"paloma","mintjulep","mojito","cremedementhe","grasshopper","curacao","bluelagoon",\
+							"bluehawaiian","negroni","necroni", "cola", "juice_lime", "juice_lemon", "juice_orange", \
+							"juice_cran", "juice_cherry", "juice_pineapple", "juice_tomato", \
+							"coconut_milk", "sugar", "water", "vanilla", "tea","mint")
+
+	var/possible_poisons = list("acetaldehyde","wolfsbane","ants","weedkiller","cyanide","krokodil", "fluorine", "radium", \
+							"neurotoxin","phlogiston","freeze","dbreath","ghostchilijuice", "cholesterol", "mercury")
```

Comment:
freeze and dbreath can easily gib and round-remove people, which is a lot more than what everything else does! I'd consider removing it, or making it require a bit more effort to unlock. Probably just remove it.